### PR TITLE
Add initial create view support

### PR DESCRIPTION
**Pull request by mfussenegger**
Note the `SQLPrinter` in here is not yet feature complete.
But it allows us to implement the full-roundtrip so that we can already work on the other parts (resolving the view in the RelationAnalyzer, information_schema extensions, etc.)

### CODE REVIEW

File: sql/src/main/java/io/crate/execution/engine/aggregation/impl/CountAggregation.java [hunk 1]
```diff
@@ -65,6 +66,24 @@ public static void register(AggregationImplModule mod) {
         mod.register(NAME, new CountAggregationFunctionResolver());
     }
 
+    @Override
+    public String beforeArgs(Function function) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/execution/engine/aggregation/impl/CountAggregation.java]: Opened a separate PR for this (https://github.com/crate/crate/pull/7043)

File: sql/src/main/java/io/crate/analyze/SQLPrinter.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            }
+        }
+
+        private void addOutputs(QueriedRelation relation, StringBuilder sb) {
+            List<Field> fields = relation.fields();
+            List<Symbol> outputs = relation.outputs();
+            for (int i = 0; i < fields.size(); i++) {
+                addOutput(sb, fields.get(i), outputs.get(i));
+
+                if (i + 1 < fields.size()) {
```
Comment (chaudum) [inline sql/src/main/java/io/crate/analyze/SQLPrinter.java]: addCommaIfNotLast() can be used here

File: sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    @Override
+    protected CreateViewResponse newResponse() {
+        return new CreateViewResponse(false);
+    }
+
+    @Override
+    protected void masterOperation(CreateViewRequest request, ClusterState state, ActionListener<CreateViewResponse> listener) {
+        ViewsMetaData views = state.metaData().custom(ViewsMetaData.TYPE);
+        if (conflictsWithTable(request.name(), state.metaData()) || conflictsWithView(request, views)) {
```
Comment (matriv) [inline sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java]: Isn't the check for "OR REPLACE" flag missing here?

Comment (matriv) [reply sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java]: ah sorry it's inside the method...

Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java]: Can't replaces tables, even if OR REPLACE is set it's an error

File: sql/src/test/java/io/crate/integrationtests/ViewsITest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            ViewsMetaData views = clusterService.state().metaData().custom(ViewsMetaData.TYPE);
+            assertThat(views, Matchers.notNullValue());
+            assertThat(views.contains(sqlExecutor.getDefaultSchema() + ".v1"), is(true));
+        }
+    }
+
+    @Test
+    public void testViewCanBeCreatedAndThenReplaced() throws Exception {
+        execute("create view v2 as select 1");
+        execute("create or replace view v2 as select 1");
```
Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/ViewsITest.java]: Shouldn't we use another query? for the 2nd (replace) statement

File: sql/src/main/java/io/crate/execution/support/OneRowActionListener.java [hunk 5]
```diff
@@ -45,7 +45,14 @@ public OneRowActionListener(RowConsumer consumer, Function<? super Response, ? e
 
     @Override
     public void onResponse(Response response) {
-        consumer.accept(InMemoryBatchIterator.of(toRowFunction.apply(response), SENTINEL), null);
+        final Row row;
```
Comment (matriv) [inline sql/src/main/java/io/crate/execution/support/OneRowActionListener.java]: Is this change related to the views?

File: sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java (hunk 3 above)
Comment (matriv) [reply sql/src/main/java/io/crate/execution/ddl/views/TransportCreateViewAction.java]: yep, yep, sorry I rushed and didn't check the methods used here.

File: sql/src/main/java/io/crate/execution/support/OneRowActionListener.java (hunk 5 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/support/OneRowActionListener.java]: Yes, see https://github.com/crate/crate/pull/7047/files/2d9392826523b6ec74cc501b9457b2a36a93c43d#diff-798d69a5d85aa135cae9d6018cc0e4f1R64

File: sql/src/test/java/io/crate/integrationtests/ViewsITest.java (hunk 4 above)
Comment (mfussenegger) [reply sql/src/test/java/io/crate/integrationtests/ViewsITest.java]: The tests generally don't verify the saved query definition yet. Will extend this once we've the information schema available

Comment (matriv) [reply sql/src/test/java/io/crate/integrationtests/ViewsITest.java]: ok, thought we can check it from ViewsMetaData but it's ok if it's gonna be replaced with "full-blown" integ test on information_schema later on.
